bigtable-client-core/src/test/java/com/google/cloud/bigtable/grpc/async/RetryingRpcFutureFallbackTest.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2015 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.cloud.bigtable.grpc.async;
+
+import static org.mockito.Mockito.when;
+import io.grpc.Status;
+
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.atomic.AtomicLong;
+
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.mockito.invocation.InvocationOnMock;
+import org.mockito.stubbing.Answer;
+
+import com.google.api.client.util.BackOff;
+import com.google.api.client.util.NanoClock;
+import com.google.bigtable.v1.ReadRowsRequest;
+import com.google.cloud.bigtable.config.RetryOptions;
+import com.google.cloud.bigtable.grpc.scanner.ScanRetriesExhaustedException;
+
+/**
+ * Test for {@link RetryingRpcFutureFallback}
+ */
+@RunWith(JUnit4.class)
+@SuppressWarnings({ "unchecked", "rawtypes" })
+public class RetryingRpcFutureFallbackTest {
+
+  private RetryingRpcFutureFallback underTest;
+
+  @Mock
+  private RetryableRpc readAsync;
+
+  @Rule
+  public ExpectedException expectedException = ExpectedException.none();
+
+  @Mock
+  private NanoClock nanoClock;
```

Comment:
I tried something, but the test wasn't at all clear in terms of intent.  I tried something a bit different.  How do you like RetryOptions.createBackoff()?